packages/dds/tree/src/object-forest/objectForest.ts

```diff
@@ -0,0 +1,292 @@
+    abstract find(forest: ObjectForest, observer: ObservingDependent | undefined): ObjectNode | undefined;
+}
+
+class RootAnchor extends ObjectAnchor {
+    find(forest: ObjectForest, observer: ObservingDependent | undefined): ObjectNode | undefined {
+        const field = forest.getRoot(forest.rootField);
+        return field[0];
+    }
+}
+
+/**
+ * Simple anchor that just points to a node object.
+ * This results in pretty basic anchor rebase policy.
+ */
+class NodeAnchor extends ObjectAnchor {
+    public constructor(public readonly node: ObjectNode) {
+        super();
+    }
+
+    find(forest: ObjectForest): ObjectNode | undefined {
+        return this.node;
+    }
+}
+
+class ObjectNode {
+    state: ITreeSubscriptionCursorState = ITreeSubscriptionCursorState.Current;
+    public readonly children: Map<FieldKey, ObjectField> = new Map();
+    public constructor(public type: TreeType, public value: Value = undefined) { }
+    free(): void {
+        assert(this.state !== ITreeSubscriptionCursorState.Freed, "Anchor must not be double freed");
+        this.state = ITreeSubscriptionCursorState.Freed;
+    }
+}
+
+type ObjectField = ObjectNode[];
+
+/**
+ * TODO: track observations.
+ */
+class Cursor implements ITreeSubscriptionCursor {
+    state: ITreeSubscriptionCursorState = ITreeSubscriptionCursorState.Cleared;
+    public constructor(public readonly forest: ObjectForest) { }
+
+    observer?: ObservingDependent | undefined;
+
+    // TODO: store stack here,
+    // then brute force on anchor restoration? (Add smarter anchor type later?)
+    private root: DetachedRange | undefined;
+
+    // Ancestors traversed to visit this node (including this node).
+    private readonly parentStack: ObjectNode[] = [];
+    // Keys traversed to visit this node
+    private readonly keyStack: FieldKey[] = [];
+    // Indices traversed to visit this node
+    private readonly indexStack: number[] = [];
+
+    private siblings?: ObjectNode[];
+
+    public clear(): void {
+        assert(this.state !== ITreeSubscriptionCursorState.Freed, "Cursor must not be freed");
+        this.root = undefined;
+        this.state = ITreeSubscriptionCursorState.Cleared;
+        this.parentStack.length = 0;
+        this.keyStack.length = 0;
+        this.indexStack.length = 0;
+        this.siblings = undefined;
+    }
+
+    public set(root: DetachedRange, index: number): void {
+        assert(this.state !== ITreeSubscriptionCursorState.Freed, "Cursor must not be freed");
+        this.clear();
+        this.root = root;
+        this.state = ITreeSubscriptionCursorState.Current;
+        this.indexStack.push(index);
+        this.siblings = this.forest.getRoot(root);
+    }
+
+    getNode(): ObjectNode {
+        assert(this.state === ITreeSubscriptionCursorState.Current, "Cursor must be current to be used");
+        assert(this.parentStack.length > 0, "Cursor must be current to be used");
+        return this.parentStack[this.parentStack.length - 1];
```

Comment:
Do we not expect to be able to call `getNode` after `set` is called? Why not?

Reply:
I expect that to work,